lib/tapyrus/bip175.rb

```diff
@@ -0,0 +1,64 @@
+module Tapyrus
+  # Key generation based on BIP-175
+  #
+  # @example
+  #
+  # master = Tapyrus::ExtKey.from_base58('xprv9s21ZrQH143K2JF8RafpqtKiTbsbaxEeUaMnNHsm5o6wCW3z8ySyH4UxFVSfZ8n7ESu7fgir8imbZKLYVBxFPND1pniTZ81vKfd45EHKX73')
+  # bip175 = Tapyrus::BIP175.from_private_key(master)
+  # bip175 << "foo"
+  # bip175 << "bar"
+  # bip175.addr
+  # > 1C7f322izqMqLzZzfzkPAjxBzprxDi47Yf
+  #
+  # @see https://github.com/bitcoin/bips/blob/master/bip-0175.mediawiki
+  class BIP175
+    PURPOSE_TYPE = 175
+
+    attr_accessor :payment_base
+
+    def initialize
+      @contracts = []
+    end
+
+    # @param key [Tapyrus::ExtKey] master private extended key
+    def self.from_private_key(key)
```

Comment:
`from_ext_key` is better because it clarify `key` is Extkey. And please add validation whether key is_a?(ExtKey).

Reply:
Rename method name (fb72383)